odoo/addons/base/models/ir_http.py

```diff
@@ -215,6 +215,8 @@ def _handle_exception(cls, exception):
     def _dispatch(cls):
         cls._handle_debug()
 
+        request.redirect = lambda url, code=302, open_redirect=False: cls._redirect(url, code=code, open_redirect=open_redirect)
```

Comment:
as discussed, define it directly inside `WebRequest` in http.py, so there is no need for dynamic monkey-patching:

```py
    # in WebRequest
    def redirect(self, location, code=303, local=True):
        if local:
            location = urls.url_parse(location).replace(scheme='', netloc='').to_url()
        if request and request.db:
            return request.registry['ir.http']._redirect(location, code)
        return werkzeug.utils.redirect(location, code, Response=Response)
```